src/libraries/System.Private.CoreLib/src/System/Span.cs

```diff
@@ -140,7 +140,7 @@ internal Span(ref T ptr, int length)
             {
                 if ((uint)index >= (uint)_length)
                     ThrowHelper.ThrowIndexOutOfRangeException();
-                return ref Unsafe.Add(ref _pointer.Value, index);
+                return ref Unsafe.Add(ref _pointer.Value, (nint)(uint)index);
```

Comment:
Yeah, this'll get intrinsicified anyway. But seemed like low-hanging fruit that could be addressed.